src/WebJobs.Script/Binding/HttpBinding.cs

```diff
@@ -31,44 +31,49 @@ public override bool HasBindingParameters
-            {
-                content = await streamReader.ReadToEndAsync();
-            }
-
-            HttpResponseMessage response = null;
-            try
+            HttpResponseMessage response = context.Value as HttpResponseMessage;
+            if (response == null)
             {
-                // attempt to read the content as a JObject
-                JObject jsonObject = JObject.Parse(content);
+                // TODO: Need to handle Stream conversions properly
+                Stream valueStream = context.Value as Stream;
+                string content;
+                using (StreamReader streamReader = new StreamReader(valueStream))
+                {
+                    content = await streamReader.ReadToEndAsync();
+                }
 
-                // TODO: This logic needs to be made more robust
-                // E.g. we might decide to use a Regex to determine if
-                // the json is a response body or not
-                if (jsonObject["status"] != null && jsonObject["body"] != null)
+                try
                 {
-                    HttpStatusCode statusCode = (HttpStatusCode)jsonObject.Value<int>("status");
-                    string body = jsonObject.Value<string>("body");
+                    // attempt to read the content as a JObject
+                    JObject jsonObject = JObject.Parse(content);
 
-                    response = new HttpResponseMessage(statusCode);
-                    response.Content = new StringContent(body);
+                    // TODO: This logic needs to be made more robust
+                    // E.g. we might decide to use a Regex to determine if
+                    // the json is a response body or not
+                    if (jsonObject["status"] != null && jsonObject["body"] != null)
+                    {
+                        HttpStatusCode statusCode = (HttpStatusCode)jsonObject.Value<int>("status");
+                        string body = jsonObject.Value<string>("body");
+
+                        response = new HttpResponseMessage(statusCode);
+                        response.Content = new StringContent(body);
+                    }
+                }
+                catch (JsonException)
+                {
+                    // not a json response
                 }
-            }
-            catch (JsonException)
-            {
-                // not a json response
-            }
 
-            if (response == null)
-            {
-                // if unable to parse a json response just send
-                // the raw content
-                response = new HttpResponseMessage
+                if (response == null)
                 {
-                    StatusCode = HttpStatusCode.OK,
-                    Content = new StringContent(content)
-                };
+                    // if unable to parse a json response just send
+                    // the raw content
+                    response = new HttpResponseMessage
```

Comment:
This wasn't part of my changes (it was probably on the initial C# scaffolding you had done) but I think it makes sense to keep it where it is or move it into the catch block where the intention may be clearer.